drivers/seafile/driver.go

```diff
@@ -51,176 +40,120 @@ func (d *Seafile) Drop(ctx context.Context) error {
+		req.SetResult(&resp).SetQueryParams(map[string]string{
 			"p": path,
-		}).
-		Get(d.Address + fmt.Sprintf("/api2/repos/%s/dir/", d.Addition.RepoId))
-
+		})
+	})
 	if err != nil {
 		return nil, err
 	}
-
-	var files []model.Obj
-	for i := 0; i < len(*resp); i++ {
-		f := (*resp)[i]
-		file := model.ObjThumb{
+	return utils.SliceConvert(resp, func(f RepoDirItemResp) (model.Obj, error) {
+		return &model.ObjThumb{
 			Object: model.Object{
 				Name:     f.Name,
 				Modified: time.Unix(f.Modified, 0),
 				Size:     f.Size,
 				IsFolder: f.Type == "dir",
 			},
 			// Thumbnail: model.Thumbnail{Thumbnail: f.Thumb},
-		}
-		files = append(files, &file)
-	}
-
-	return files, nil
+		}, nil
+	})
 }
 
 func (d *Seafile) Link(ctx context.Context, file model.Obj, args model.LinkArgs) (*model.Link, error) {
-	resp, err := base.RestyClient.R().
-		SetHeader("Authorization", d.authorization).
-		SetQueryParams(map[string]string{
+	res, err := d.request(http.MethodGet, fmt.Sprintf("/api2/repos/%s/file/", d.Addition.RepoId), func(req *resty.Request) {
+		req.SetQueryParams(map[string]string{
 			"p":     file.GetPath(),
 			"reuse": "1",
-		}).
-		Get(d.Address + fmt.Sprintf("/api2/repos/%s/file/", d.Addition.RepoId))
-
+		})
+	})
 	if err != nil {
 		return nil, err
 	}
-
-	u := resp.String()
+	u := string(res)
 	u = u[1 : len(u)-1] // remove quotes
-
 	return &model.Link{URL: u}, nil
 }
 
 func (d *Seafile) MakeDir(ctx context.Context, parentDir model.Obj, dirName string) error {
-	resp, err := base.RestyClient.R().
-		SetHeader("Authorization", d.authorization).
-		SetQueryParams(map[string]string{
+	_, err := d.request(http.MethodPost, fmt.Sprintf("/api2/repos/%s/dir/", d.Addition.RepoId), func(req *resty.Request) {
+		req.SetQueryParams(map[string]string{
 			"p": filepath.Join(parentDir.GetPath(), dirName),
-		}).
-		SetFormData(map[string]string{
+		}).SetFormData(map[string]string{
 			"operation": "mkdir",
-		}).
-		Post(d.Address + fmt.Sprintf("/api2/repos/%s/dir/", d.Addition.RepoId))
-
-	if resp.StatusCode() == 201 {
-		return nil
-	}
-
+		})
+	})
 	return err
 }
 
 func (d *Seafile) Move(ctx context.Context, srcObj, dstDir model.Obj) error {
-	resp, err := base.NoRedirectClient.R().
-		SetHeader("Authorization", d.authorization).
-		SetQueryParams(map[string]string{
+	_, err := d.request(http.MethodPost, fmt.Sprintf("/api2/repos/%s/file/", d.Addition.RepoId), func(req *resty.Request) {
+		req.SetQueryParams(map[string]string{
 			"p": srcObj.GetPath(),
-		}).
-		SetFormData(map[string]string{
+		}).SetFormData(map[string]string{
 			"operation": "move",
 			"dst_repo":  d.Addition.RepoId,
 			"dst_dir":   dstDir.GetPath(),
-		}).
-		Post(d.Address + fmt.Sprintf("/api2/repos/%s/file/", d.Addition.RepoId))
-
-	if resp.StatusCode() == 301 {
-		return nil
-	}
-
+		})
+	})
 	return err
 }
 
 func (d *Seafile) Rename(ctx context.Context, srcObj model.Obj, newName string) error {
```

Comment:
Does this refactor keep not follow to the 301 redirect?

Because of the Seafile problem, when serving on non 80 port behind a reverse proxy, the successful of the `rename` API will redirect our request to new uri but with 80 port, it cause TCP dail timeout.

Reply:
@Xhofe

Reply:
Sorry I seem to have overlooked this